comments/parse_docs.go

```diff
@@ -91,29 +88,27 @@ func parseFunctions(fset *token.FileSet, pkg *types.Package, decl *ast.GenDecl,
 			return nil, fmt.Errorf("must have one name")
 		}
 		name := value.Names[0].Name
-
-		location := fileWithLine(fset.Position(value.Pos()))
-		result[name] = parseRawLines(location, value.Doc.Text())
+		result[name] = rawLines(pkg, value)
 	}
 
 	converter := config.RawConverter{
 		FileName:    location.Filename,
-		Converter:   converterLines,
+		Converter:   lines,
 		Methods:     result,
-		PackageName: pkg.Name(),
-		PackagePath: pkg.Path(),
+		PackageName: pkg.Name,
+		PackagePath: pkg.PkgPath,
 	}
 	return []config.RawConverter{converter}, nil
 }
 
-func parseGenDecl(fset *token.FileSet, pkg *types.Package, decl *ast.GenDecl) ([]config.RawConverter, error) {
-	declDocs := decl.Doc.Text()
```

Comment:
The CommentGroup can be obtained directly from decl.Doc. This should remove the need to nearly all changes to this file. The SettingLines method should be used again, and we have a new method that does what CommentGroup.Text() does, but without skipping directives, and then we only do `SettingLines(CommentToString(CommentGroup))`.

https://github.com/golang/go/blob/master/src/go/ast/ast.go#L97

The adjusted Text() method should be in a separate file with the signature `func(CommentGroup) string` and it does only the to string part not any other parsing.

Reply:
That's almost exactly what I had on my first pass except I called it `CommentText` instead of `CommentToString`. Then I started thinking that there has to be a more "official" way to do this than modifying a stdlib func and ended up with the monstrosity I submitted.